### ISSUE TITLE
File object opened via SSH is inconsistent with the file object API

### ISSUE THREAD

**Issue by mxmlnkn** [fsspec/filesystem_spec#1695]
This is kind of a duplicate of the [paramiko issue](https://github.com/paramiko/paramiko/issues/2452). But, I think it could also be fixed in fsspec. I'm not sure whether fsspec does define a standard for the file object. If not, maybe it should? I find it highly unexpected that for the file object:

 - seek does not return the new position
 - write does not return the number of bytes written
 - probably other issues?

This could be fixed here instead of upstream by adding a small wrapper around the paramiko file object. Of course, the same [discussion](https://github.com/fsspec/filesystem_spec/issues/1683#issuecomment-2371738666) about fsspec/sshfs applies here.

**Comment by martindurant** [comment]
I'd be happy to see the file object wrapped.

> the same discussion about fsspec/sshfs applies

Right, any fixes here might end up waster effort. I take it sshfs already fills out the file API?

**Comment by mxmlnkn** [comment]
> I take it sshfs already fills out the file API?

Yes, it [inherits from IOBase](https://github.com/fsspec/sshfs/blob/15aac74de1d2e1c28376e70211397a8533318977/sshfs/file.py#L14) and then simply redirects to the asynchssh's [seek](https://github.com/ronf/asyncssh/blob/1f9c8abeaaf32abb0c3c1aa36924245d2402e984/asyncssh/sftp.py#L3340) and [write](https://github.com/ronf/asyncssh/blob/1f9c8abeaaf32abb0c3c1aa36924245d2402e984/asyncssh/sftp.py#L3289), which are sufficiently compatible.
